### PR TITLE
code cleanup

### PR DESCRIPTION
**Pull request by muskeinsingh**
Code cleanup. Importing from RecyclerListView rather than keeping a clone.
Resolving dependencies for development with RecyclerListView.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -34,7 +34,7 @@
     "url": "https://github.com/muskeinsingh/recyclerlistview-gridlayoutmanager/issues"
   },
   "homepage": "https://github.com/muskeinsingh/recyclerlistview-gridlayoutmanager",
-  "peerDependencies": {
-    "recyclerlistview": ">= 1.4.0"
+  "dependencies": {
```
Comment (naqvitalha) [inline package.json]: keep as peer.. also add to dev to resolve ts errors

File: scripts/build.sh [hunk 2]
```diff
@@ -6,7 +6,10 @@ tslint 'src/**/*.{ts,tsx}'
 
 echo "Build started..."
 echo "Removing old builds..."
+rm -rf node_modules
```
Comment (naqvitalha) [inline scripts/build.sh]: There is a clean script already

File: src/GridLayoutProvider.ts [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+  private _getDimensionForIndex: (index: number) => number;
+  private _getSpanForIndex: (index: number) => number;
+  private _setMaxSpan: () => number;
+  private _renderWindowSize: Dimension | undefined;
+  private _isHorizontal: boolean | undefined;
+  constructor(
+    getLayoutTypeForIndex: (index: number) => string | number,
+    getDimensionForIndex: (index: number) => number,
+    getSpanForIndex: (index: number) => number,
+    setMaxSpan: () => number,
```
Comment (naqvitalha) [inline src/GridLayoutProvider.ts]: Why is this still a method?

**Review by naqvitalha — CHANGES_REQUESTED** [review]

File: package.json [hunk 4]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "recyclerlistview-gridlayoutmanager",
```
Comment (naqvitalha) [inline package.json]: call it gridlayoutprovider

**Review by naqvitalha — CHANGES_REQUESTED** [review]

File: src/GridLayoutProvider.ts (hunk 3 above)
Comment (muskeinsingh) [reply src/GridLayoutProvider.ts]: This is for the GridLayoutProvider. GridLayoutManager gets the value in the constructor directly.

**Review by naqvitalha — APPROVED** [review]

**Pull request merged**